File: modules/lift/src/main/_java/org/scalabox/lift/extension/TypeRemoveHandler.java
package org.scalabox.lift.extension;

import org.jboss.as.controller.AbstractRemoveStepHandler;
import org.jboss.as.controller.OperationContext;
import org.jboss.as.controller.OperationFailedException;
import org.jboss.as.controller.PathAddress;
import org.jboss.as.controller.descriptions.DescriptionProvider;
import org.jboss.dmr.ModelNode;
import org.jboss.msc.service.ServiceName;

import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.*;

import java.util.Locale;

/**
 * // TODO: Document this
 *
 * @author Galder Zamarreño
 * @since // TODO
 */
public class TypeRemoveHandler extends AbstractRemoveStepHandler
      implements DescriptionProvider {

   public static final TypeRemoveHandler INSTANCE = new TypeRemoveHandler();

   private TypeRemoveHandler() {
   }

   @Override
   public ModelNode getModelDescription(Locale locale) {
      ModelNode node = new ModelNode();
      node.get(DESCRIPTION).set("Removes a Lift deployment type");
      return node;
   }

   @Override
   protected void performRuntime(OperationContext context,
               ModelNode operation, ModelNode model)
         throws OperationFailedException {
      String suffix = PathAddress.pathAddress(
            operation.get(ADDRESS)).getLastElement().getValue();
      ServiceName name = LiftService.createServiceName(suffix);
      context.removeService(name);
   }

}

File: modules/lift/src/main/scala/org/scalabox/lift/extension/TypeRemoveHandler.scala
package org.scalabox.lift.extension

import org.jboss.as.controller.descriptions.DescriptionProvider
import java.util.Locale
import org.jboss.dmr.ModelNode
import org.jboss.as.controller.descriptions.ModelDescriptionConstants._
import org.jboss.as.controller.{OperationContext, PathAddress, AbstractRemoveStepHandler}

/**
 * // TODO: Document this
 * @author Galder Zamarreño
 * @since // TODO
 */
object TypeRemoveHandler extends AbstractRemoveStepHandler with DescriptionProvider {

   def getModelDescription(locale: Locale): ModelNode = {
      val node = new ModelNode
      node.get(DESCRIPTION).set("Removes a Lift deployment type")
      node
   }

   protected override def performRuntime(ctx: OperationContext,
            op: ModelNode, model: ModelNode){
      val suffix = PathAddress.pathAddress(op.get(ADDRESS)).getLastElement.getValue
      val name = LiftService.createServiceName(suffix)
      ctx.removeService(name)
   }

}